sentinel-ai-models-simple-openai/src/main/java/com/phonepe/sentinelai/models/SimpleOpenAIModel.java

```diff
@@ -476,6 +483,22 @@ yield processStreamingOutput(context,
         }, agentSetup.getExecutorService());
     }
 
+    private static Optional<ModelOutput> toModelOutput(final ModelRunContext context,
+                                                       final Throwable error,
+                                                       final List<AgentMessage> newMessages,
+                                                       final List<AgentMessage> allMessages) {
+        var rootCause = AgentUtils.rootCause(error);
+        if (rootCause instanceof SocketTimeoutException
+                || rootCause instanceof SocketException) {
```

Comment:
SocketException might not be timeout. Could be connection refused or connection closed etc as well no? maybe we should add some more error codes and close the relevant TODOs